pkg/daemon/daemon.go

```diff
@@ -129,6 +129,11 @@ type Daemon struct {
 	configDriftMonitor ConfigDriftMonitor
 }
 
+// CoreOSDaemon protects the methods that should only be called on CoreOS variants
```

Comment:
I think we're definitely going to want more variants. At least (Legacy)Daemon, (Legacy)CoreOSDaemon, and LayeredDaemon. I've thought briefly about what that could look like but just haven't put the effort into coming up with something that works yet. It is a little tricky because the interface can't just be `update()` because a legacy update would need `update(oldMachineConfig, newMachineConfig)`, and (I don't think) the MachineConfigs are always calculated the same way for current code paths. Whereas a layered update will just need the image annotation.

Reply:
Yeah, that makes sense. There's definitely more stuff to flesh out than my admittedly-naïve example includes. But in general, it would be nice to have a common interface for all the variants since each variant can have its own independent implementation so we don't have a bunch of duplicated logic.

Reply:
I think it's fine for now. I do want to figure out why the CI runs failed first before this gets merged.